consensus_test.go

```diff
@@ -373,9 +372,21 @@ func TestTransition_RoundChangeState_CatchupRound(t *testing.T) {
 func TestTransition_RoundChangeState_Timeout(t *testing.T) {
 	m := newMockPbft(t, []string{"A", "B", "C", "D"}, "A")
 
-	m.forceTimeout()
 	m.setState(RoundChangeState)
-	m.Close()
+
+	// Jump out from a state machine loop straight away after round 2
+	waitSignal := make(chan struct{})
```

Comment:
cancelFn was called before runCycle, but since forceTimeout was true, on getNextMessage, it did not go to select where it listened to timeout, update and cancel channels. It just returned false. Now, since forceTimeout is removed, and since we have mockPbft where is round timeout is 1ms, when round two is reached, it will have no more messages to read and a timeout will be invoked. But, since we called cancelFn before runCycle, its channel will already have a value in it, and timeout will never be reached. That is why we needed to have this 'hack' to cancel runCycle after timeout.